### DISCUSSION TITLE
Compiling Custom Builds for CR4NA200141C13 Board (STM32F401RET6)

### DISCUSSION THREAD

**Discussion by okokel** [mriscoc/Ender3V2S1#1115] [Q&A]
I love this firmware, to the point where I've delayed switching to klipper because of how well my Ender-3 S1 and Ender-3 V2 are working right now using custom versions of the latest build.  However, the only downside has been the limited features I can add to my S1 due to the smaller program memory (STM32F401RCT6).  I recently realized that the Ender-5 S1 board (Creality CR4NS200141C13) looks almost identical to the Ender-3 S1/S1 Pro Board (V2.4.S1.301), with the notable difference of having an STM32F401RET6 chip instead.  I'm curious if anyone has attempted to load either precompiled firmware or custom versions using the Ender-3 S1 settings onto a CR4NS200141C13 board.  If additional coding is not needed, this would be the simplest way around the program size limitation for me; I'd just need to use the RET6 option when compiling with Marlin Auto Build.  If it isn't directly compatible and would require some tweaks to the marlin setup files, what would be required to achieve functionality?  I did see a couple threads on this and other forums with people saying they have a CRNS200141C13 board with RCT6 chip in their S1 Pro, but haven't seen enough cases to have confidence based on their experiences.

Links to Microcenter's product pages for the boards in question.  At the time of writing, there are two listings for the Ender-3 S1, one for the non-Pro (out of stock, V2.4.S1.301), and the Pro (image shows CRNS200141C13 with RCT6 chip).  Assuming Microcenter's images are to be believed, lends credence to them being interchangeable.

Ender-5 S1 (CRNS200141C13 RET6): https://www.microcenter.com/product/667170/creality-ender-5-s1-silent-mainboard-v13-32-bit-with-tmc2208-driver

Ender-3 S1 (V2.4.S1.301): https://www.microcenter.com/product/659609/creality-mainboard-for-ender-3-s1?ob=1

Ender-3 S1 Pro (CRNS200141C13 RCT6): https://www.microcenter.com/product/667152/creality-ender-3-s1-pro-silent-mainboard

**Comment by charliechan84** [comment]
same issue; just got mine the other day and unfortunately there isnt any aftermarkets that work sadly, so i had to go back to stock firmware for it to even work

**Comment by mriscoc** [comment]
Check here: https://github.com/mriscoc/Ender3V2S1/discussions/745#discussioncomment-5437014
